reconcile/change_owners/change_owners.py

```diff
@@ -120,7 +121,71 @@ def fetch_bundle_changes(comparison_sha: str) -> list[BundleFileChange]:
     explicitely passed comparision bundle - usually the state of the master branch).
     """
     changes = gql.get_diff(comparison_sha)
-    return _parse_bundle_changes(changes)
+    bundle_changes = _parse_bundle_changes(changes)
+    try:
+        return _aggregate_file_moves(bundle_changes)
+    except Exception as e:
+        logging.error(f"Failed to post process bundle changes: {e}")
+        # if any post processing action fails, we want the raw bundle changes
+        # to be reviewed
+        return bundle_changes
+
+
+def _aggregate_file_moves(
+    bundle_changes: list[BundleFileChange],
+) -> list[BundleFileChange]:
+    """
+    This function tries to detect file moves by looking at the bundle changes. If an
+    add and remove file change with the same content is detected, those changes are
+    replaced with a single move change where the only difference is the path.
+    """
+    move_candidates: dict[str, list[BundleFileChange]] = defaultdict(list)
+    new_bundle_changes = []
+    for c in bundle_changes:
+        potential_move_sha = None
+        if c.is_file_creation():
+            potential_move_sha = c.new_content_sha()
+        elif c.is_file_deletion():
+            potential_move_sha = c.old_content_sha()
+        if potential_move_sha:
+            move_candidates[potential_move_sha].append(c)
+        else:
+            new_bundle_changes.append(c)
+
+    for candidates in move_candidates.values():
+        if len(candidates) == 2:
+            # if there are two candidates, they could represent a file move.
+            # lets check if that is the case
+            deletion = (
+                candidates[0] if candidates[0].is_file_deletion() else candidates[1]
+            )
+            addition = (
+                candidates[0] if candidates[0].is_file_creation() else candidates[1]
+            )
```

Comment:
Nit: these checks have duplicate logic with init step, to make it simpler and less nested, we can extract a dataclass

```python
@dataclass
class MoveCandidates:
    creations: list[BundleFileChange] = field(default_factory=list)
    deletions: list[BundleFileChange] = field(default_factory=list)

    def changes(self) -> list[BundleFileChange]:
        if len(self.creations) == 1 and len(self.deletions) == 1:
            creation = self.creations[0]
            deletion = self.deletions[0]
            if (
                creation != deletion
                and deletion.fileref.file_type == creation.fileref.file_type
                and deletion.fileref.path != creation.fileref.path
            ):
                move_change = create_bundle_file_change(
                    deletion.fileref.path,
                    deletion.fileref.schema,
                    deletion.fileref.file_type,
                    deletion.old,
                    creation.new,
                )
                if move_change:
                    return [move_change]

        return self.creations + self.deletions
```

Then use it

```python
    move_candidates: dict[str, MoveCandidates] = defaultdict(MoveCandidates)
    new_bundle_changes = []
    for c in bundle_changes:
        if c.is_file_creation():
            move_candidates[c.new_content_sha()].creations.append(c)
        elif c.is_file_deletion():
            move_candidates[c.old_content_sha()].deletions.append(c)
        else:
            new_bundle_changes.append(c)
    move_candidate_changes = itertools.chain.from_iterable(
        c.changes() for c in move_candidates.values()
    )
    new_bundle_changes.extend(move_candidate_changes)
    return new_bundle_changes
```